app/controllers/taxons_controller.rb

```diff
@@ -93,13 +93,47 @@ def confirm_delete
     }
   end
 
+  def restore
+    Taxonomy::PublishTaxon.call(taxon: taxon)
+    redirect_to taxons_path, flash: flash_message(type: 'restore', response_code: response_code)
+  rescue Taxonomy::PublishTaxon::InvalidTaxonError => e
+    redirect_to taxons_trash_path, flash: { danger: e.message }
+  end
+
 private
 
-  def destroy_flash_message(response_code)
+  def search_and_render_taxons(deleted: false)
```

Comment:
I think it might be easier to read if we don't extract this code out into a method, but accept a bit of duplication. This method already contains two `if` statements, which we would get rid of putting it back into the action methods. 

The same goes for the flash messages - if they were inlined there would be less code to parse.